### PR TITLE
Don's edits

### PR DESCRIPTION
**Pull request by UncannyBingo**
I have edited these two articles for grammar. I have some additional thoughts on the presentation of the ideas which I have tried _not_ to do anything with in this PR, as they are better handled via talking about it ;)

### CODE REVIEW

File: _practices/safe-password-storage/python.md [hunk 1]
```diff
@@ -3,23 +3,21 @@ title: Safe password storage
...
+Don't store passwords in clear-text. Don't hash passwords with MD5. Use dedicated algorithms for hashing passwords. Use a distinct salt per user.
 
 # Vulnerable code
 
-It’s universally acknowledged that it’s a bad idea to store plain-text passwords. The recommandation is to hash them. Here are some examples of **WHAT TO NOT DO**.
+It’s universally acknowledged that storing clear-text passwords is a bad idea. The recommendation is to hash them, using a cryptographically secure hashing algorithm. Here are some examples of **WHAT TO NOT DO**.
 
 ```python
-from hashlib import md5
+from hashlib import MD5
```
Comment (Lothiraldan) [inline _practices/safe-password-storage/python.md]: Here, it's python code and there is no MD5 function in hashlib module

File: _practices/safe-password-storage/python.md [hunk 2]
```diff
@@ -3,23 +3,21 @@ title: Safe password storage
...
 
 def hash_password(password):
-    return md5(password)
+    return MD5(password)
```
Comment (Lothiraldan) [inline _practices/safe-password-storage/python.md]: Same here, please revert

File: _practices/safe-password-storage/python.md [hunk 3]
```diff
@@ -37,89 +35,89 @@ def hash_password(password):
...
+The problem with brute-forcing passwords is that you need to have a list of "known" passwords in the first place. It's called a `wordlist` and is basically a clear-text file of passwords found in previous hacks, usually the most common ones like `123456` or `qwerty`. The problem is that most users uses these simple password or a combination of them. Security researchers estimate thatas many as [50% of passwords found in leaks are from the top 25% most common passwords in 2016](https://blog.keepersecurity.com/2017/01/13/most-common-passwords-of-2016-research-study/). Passwords crackers can also apply some mangling rules, like replacing a letter by a number `qw3rty` or mixing the case `QwErTy`. So, even with a strong hash, attackers can determine the passwords of many accounts in your system.
```
Comment (Lothiraldan) [inline _practices/safe-password-storage/python.md]: typo here: thatas

**Review by Lothiraldan — CHANGES_REQUESTED** [review]
Thank you for the PR, I have two / three small comments

File: _practices/safe-password-storage/python.md (hunk 3 above)
Comment (UncannyBingo) [reply _practices/safe-password-storage/python.md]: Excellent! Addressed.

File: _practices/safe-password-storage/python.md (hunk 1 above)
Comment (UncannyBingo) [reply _practices/safe-password-storage/python.md]: Addressed

File: _practices/safe-password-storage/python.md (hunk 2 above)
Comment (UncannyBingo) [reply _practices/safe-password-storage/python.md]: Addressed

**Review by Lothiraldan — APPROVED** [review]

**Comment by Lothiraldan** [comment]
Thank you @DEGoodmanWilson for your contribution!

**Pull request merged**